receiver/googlecloudmonitoringreceiver/internal/metrics_conversion.go

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
+
+import (
+	"log"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+)
+
+type MetricsBuilder struct {
+	logger *zap.Logger
+}
+
+func NewMetricsBuilder(logger *zap.Logger) *MetricsBuilder {
+	return &MetricsBuilder{
+		logger: logger,
+	}
+}
+
+func (mb *MetricsBuilder) ConvertGaugeToMetrics(ts *monitoringpb.TimeSeries, m pmetric.Metric) pmetric.Metric {
+	m.SetName(ts.GetMetric().GetType())
+	m.SetUnit(ts.GetUnit())
+	gauge := m.SetEmptyGauge()
+
+	for _, point := range ts.GetPoints() {
+		dp := gauge.DataPoints().AppendEmpty()
+		dp.SetStartTimestamp(pcommon.Timestamp(point.Interval.StartTime.Seconds))
```

Comment:
Yes, that's correct. We can add a conditional statement: if the value exists, we set it; otherwise, we leave it unset.

Do we need to do the same thing for endtime as well ?

Reply:
Yes. I would do it just for safety.  It shouldn't ever happen, so you can log a warning and skip the point if no end time is set.